SC_redesign/Crop_and_Soil/soil/soil_erosion.py

```diff
@@ -0,0 +1,251 @@
+        self.data = soil_data or SoilData()  # Initialize with defaults, if not given
+
+    def erode(self, minimum_cover_management_factor: float, surface_residue: float) -> None:
+        """this is the main routine for SoilErosion. It is responsible for running all the necessary soil erosion
+            methods and updating attributes as necessary
+
+        Args:
+            minimum_cover_management_factor: minimum value for cover and management factor for water erosion applicable
+                to land cover/plant (unitless)
+            surface_residue: amount of residue on the soil surface (kg per hectare)
+
+        """
+        erodibility_factor = self._determine_soil_erodibility_factor(self.data.percent_sand_content,
+                                                                     self.data.percent_silt_content,
+                                                                     self.data.soil_layers[0].percent_clay_content,
+                                                                     self.data.soil_layers[0]
+                                                                     .percent_organic_carbon_content)
+        cover_factor = self._determine_cover_management_factor(minimum_cover_management_factor, surface_residue)
+        support_practice_factor = self._determine_support_practice_factor()
+        topographic_factor = self._determine_topographic_factor(self.data.slope_length,
+                                                                self.data.average_slope_fraction)
+        fragment_factor = self._determine_coarse_fragment_factor(self.data.percent_rock_content)
+
+        if self.data.peak_runoff_rate is None:
+            raise TypeError("SoilData peak_runoff_rate cannot be NoneType")
+            return
+        elif self.data.surface_runoff_volume is None:
+            raise TypeError("SoilData surface_runoff_rate cannot be NoneType")
+            return
+        sediment_yield = self._determine_sediment_yield(self.data.surface_runoff_volume, self.data.peak_runoff_rate,
+                                                        self.data.field_size, erodibility_factor, cover_factor,
+                                                        support_practice_factor, topographic_factor, fragment_factor)
+        self.data.eroded_sediment += self._determine_adjusted_sediment_yield(sediment_yield,
+                                                                             self.data.snow_cover_water_content)
+
+    # --- Static methods ---
+    @staticmethod
+    def _determine_coarse_sand_factor(percent_sand_content: float, percent_silt_content: float) -> float:
+        """calculates factor based on sand content for use in determining soil erodibility factor
+
+        Args:
+            percent_sand_content: percent of soil content that is sand
+            percent_silt_content: percent of soil content that is silt
+
+        Returns:
+            factor based on sand content (unitless)
```

Comment:
Can we describe this as something other than just "factor". Perhaps something like the following

```suggestion
        """calculates the coarseness factor of soil erodbility

        Args:
            percent_sand_content: percent of soil content that is sand
            percent_silt_content: percent of soil content that is silt
            
        Details: 
          The coarseness of a soil effects the overall erodibility of the soil. Specifically, soils with high-levels of 
          coarse-sand content will have relatively low erodibility compared to soils with less sand. 

        Returns:
            coarseness factor of erodibility, based on sand content (unitless)
```